rfd/0148-pinned-unified-resources.md

```diff
@@ -0,0 +1,149 @@
+	}
+
+	return resources, nil
+}
+```
+After this, any existing filters in the request will be applied the same (including RBAC).  
+
+### Space limitations for UserPreferences storage
+> The maximum item size in DynamoDB is **400 KB**, which includes both attribute name binary length (UTF-8 length) and attribute value lengths (again binary length). The attribute name counts towards the size limit.
+
+If we assume an average resource ID is something like `db-name-1aaa8584-0e54-4c89-bec9-34f957512078`, then we can
+store well above 10,000 pinned resources per user. This is a very unlikely scenario as any amount of pinned resources over 20, 
+lets just say for conversation sake, defeats the purpose of pinning a resource in the first place. We don't expect anyone to pin 
+more than a "page" worth. We can still limit the resources in the backend to a total (per cluster) of something like 500. 
+These are knobs we can easily turn if necessary but it seems unlikely to "deliberately" go over this cap.
+
+### "What happens if a resource I have pinned becomes unavailable?"
+Similarly to the normal resource view, if a resource becomes unavailable (due to RBAC or being removed) it just won't be visible in the pinned view either. 
+
+### Manual Cleanup
+As of now, there would be no way to manually clean up a pinned resource that isn't available because the UI would never show the tile. We could perhaps decide to show a list of tiles that aren't available but without knowing the "why" (is it gone? connectivity? rbac?) it may be more confusing to see a tile unavailable with no other info, other than the ID, than it would to just have it not shown. An example of that would be like so
+
+![Untitled-2022-09-11-1530](https://github.com/gravitational/teleport/assets/5201977/e52c4286-bf57-49cc-bfb5-d541146f6896)
+
+
+After discussing with the UX team, having the resource not show is the preferred option. Maybe a user preferences page in the future
+could list the IDs you currently have stored and allow you to remove them that way. This would not be in the first iteration.
+
+### Automatic Cleanup
+The case in which this cap _could_ be reached is when we allow unavailable/unauthorized/unreachable resources to exist and fester in a user's preferences. (lets discuss)
+
+Scenarios that would make a pinned resource ID stale
+1. User loses access to the resource. In that case, I'm not sure we'd want to remove the resource from their pinned. I could see 
+pinning a specific resource after assuming a role, and only caring about that resource's pin when you've assumed the role again.
+2. Resource loses connectivity. We probably wouldn't want to remove a resource's pin due to connection issues unless x amount of time
+has passed. 
+3. Resource changes it's name. Idk how frequent this happens.
+4. ...?
+
+### Security Concerns
+Pinned resources go through the same RBAC as unified resources so no additional security concerns matter in the listing. 
+
+### Backward compatibility
```

Comment:
We can go the route of Unified Resources support checking and just have the pinning UI (the push pin and tab) not be displayed if the cluster doesn't support pinning resources. Or, we could show an error anytime they try to pin along the lines of "yo you cant do that yet"

but I think the first option is better for UX. might as well not show them an error they (most users) won't have control over in terms of versioning instead of informing them they are wrong. we just dont _let_ them be wrong!